### PR TITLE
(MODULES-3285) Remove Beaker-rspec from this test suite

### PR DESCRIPTION
**Pull request by Iristyle**
Intent is to be more similar to other test suites not running `beaker-rspec`, so that the tests can be invoked using a simpler command line like:

```
bundle exec beaker --config tests/configs/windows-2012r2-x86_64.yml --debug --tests tests/acceptance/tests --keyfile /Users/Iristyle/.ssh/id_rsa-acceptance --pre-suite tests/acceptance/pre-suite
```

### CODE REVIEW

File: spec/acceptance/tests/command_supported_throw.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  }
+MANIFEST
+
+agents.each do |agent|
+  opts = {
+    # TODO: with this on like before, the test doesn't work right
+    # :catch_failures => true,
+    :future_parser => (ENV['FUTURE_PARSER'] != 'false'),
+    # TODO: is this the right exit code?? maybe remove this given catch_failures => true
+    :acceptable_exit_codes => [1]
```
Comment (Iristyle) [inline spec/acceptance/tests/command_supported_throw.rb]: Have you seen any quirky behavior here @cowofevil ?

Comment (cowofevil) [inline spec/acceptance/tests/command_supported_throw.rb]: No, this is SOP for writing acceptance in QA. The only comment I would make is that the exit code doesn't need to be in an array if specifying only one.

Comment (cowofevil) [inline spec/acceptance/tests/command_supported_throw.rb]: I feel like a better option would be to use the ```:expect_failures``` option. See [Beaker docs for more details](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Helpers/PuppetHelpers:apply_manifest_on).

File: spec/acceptance/tests/env_var_existing_accessible.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+test_name 'PowerShell Module - should see environment variables set outside of session'
```
Comment (Iristyle) [inline spec/acceptance/tests/env_var_existing_accessible.rb]: FYI @cowofevil you can see how I rewrote this test

Comment (cowofevil) [inline spec/acceptance/tests/env_var_existing_accessible.rb]: Setting the variable outside of PowerShell is a good idea. Also, I really like the pattern you use for ```apply_manifest_on```. It makes it very readable.

File: spec/acceptance/tests/command_supported_throw.rb (hunk 1 above)
Comment (Iristyle) [inline spec/acceptance/tests/command_supported_throw.rb]: Yeah, was looking through that already... wanted your opinion on idiomatic test writing for QA.  Will change per your recommendation.

Thanks

**Comment by Iristyle** [comment]
@ThoughtCrhyme I'll let you run through these tests to verify that the behavior all makes sense / tests are being verified properly.

**Comment by Iristyle** [comment]
For this to work properly we will first need to:

* Verify related CI changes for this pipeline in a staging area
* Commit and roll out those changes to the production pipeline

@glennsarti is going to take care of that bit, and come back here with a 👍 when it's ready to go.

**Comment by glennsarti** [comment]
Adhoc CI pipeline threw an error

Appears that the tests/configs node files are not named correctly

```
/var/lib/jenkins/workspace/forge-windows_puppetlabs-powershell_intn-sys_smoke-adhoc/PLATFORM/windows-2012r2-64a/WORKER_LABEL/beaker/tests/configs/windows-2012r2-64a' does not exist!
```
Using beaker invocation of
```
bundle exec beaker                       \
  --debug                                \
  --preserve-hosts never                 \
  --config tests/configs/$PLATFORM       \
  --keyfile ~/.ssh/id_rsa-acceptance     \
  --pre-suite tests/acceptance/setup     \
  --tests tests/acceptance/tests
```

**Comment by Iristyle** [comment]
Thanks @glennsarti - I had noticed that difference in the filenames with the DSC pipeline, but hadn't put 2 and 2 together.

I think we will also want to change the platform names to follow `beaker-hostgenerator` conventions - i.e. masterless vs not.

Will update PR accordingly once I can dig up the proper renames for everything.

**Comment by glennsarti** [comment]
May be best to reword the commit for `Remove ENV['FUTURE_PARSER']`  I read that as FUTURE_PARSER itself is no longer required instead of `Beaker already does this and is no longer requried`

File: tests/acceptance/pre-suite/02-install_powershell_module.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+proj_root = File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..'))
+staging = { :module_name => 'puppetlabs-powershell' }
+local = { :module_name => 'powershell', :source => proj_root }
+
+hosts.each do |host|
+  step 'Install PowerShell Module'
+  # in CI allow install from staging forge, otherwise from local
+  install_dev_puppet_module_on(host, options[:forge_host] ? staging : local)
+end
```
Comment (Iristyle) [inline tests/acceptance/pre-suite/02-install_powershell_module.rb]: Looks like this file wasn't git moved... and is missing its end of file terminator.

Comment (glennsarti) [inline tests/acceptance/pre-suite/02-install_powershell_module.rb]: Yes it's a new file.

**Comment by glennsarti** [comment]
@Iristyle Can you review my commits and then we can get this merged

**Comment by Iristyle** [comment]
A couple of additional notes on this one:

* May no longer need to worry about passing around a future parser config flag
* This should be undoing https://github.com/puppetlabs/puppetlabs-powershell/commit/8070cdd43e815c6adb286fcfd95a93b66aa75428
* I believe that I was executing this suite with:

```
BEAKER_PE_DIR=http://pe-releases.puppetlabs.lan/2016.1.1/ bundle exec beaker --preserve-hosts onfail --config spec/acceptance/nodesets/windows-2012r2-64a --debug --tests spec/acceptance --keyfile /Users/Iristyle/.ssh/id_rsa-acceptance --load-path spec
```

Or maybe?

```
BEAKER_PE_DIR=http://pe-releases.puppetlabs.lan/2016.1.1/ BEAKER_setfile=spec/acceptance/nodesets/windows-2012r2-x86_64.yml BEAKER_keyfile=/Users/Iristyle/.ssh/id_rsa-acceptance BEAKER_debug=true BEAKER_destroy=no bundle exec rake beaker
```

**Comment by glennsarti** [comment]
Closing as this won't be done.

**Pull request closed without merging**